packages/packagers/html/src/HTMLPackager.js

```diff
@@ -112,6 +114,51 @@ async function getAssetContent(
   return null;
 }
 
+async function addIntrinsicImageSizes(
+  bundle: NamedBundle,
+  bundleGraph: BundleGraph<NamedBundle>,
+  tree,
+) {
+  const imageNodes = [];
+  tree.walk(node => {
+    if (node.tag === 'img' && node.attrs && node.attrs['src']) {
+      imageNodes.push(node);
+    }
+    return node;
+  });
+
+  for (const node of imageNodes) {
+    const dependency = bundle
+      .getMainEntry()
+      ?.getDependencies()
+      .find(bundle => bundle.id === node.attrs['src']);
+
+    if (!dependency) {
+      continue;
+    }
+
+    const image = bundleGraph.getResolvedAsset(dependency);
+
+    if (!image) {
+      continue;
+    }
+
+    if (
+      typeof image.meta.width !== 'number' &&
+      typeof image.meta.height !== 'number'
+    ) {
+      const size = imageSize(await image.getBuffer());
+      image.meta.width = size.width;
+      image.meta.height = size.height;
```

Comment:
I'm not sure we should mutate assets in packagers. I'm actually surprised this works... 😬

Reply:
It could be that the image transformer sets the sizes (even if it's not actually transforming the image), and this sets them if present.